### ISSUE TITLE
[FEATURE] add databricks_workspace data source to retrieve an existing workspace

### ISSUE THREAD

**Issue by riturr** [databricks/terraform-provider-databricks#1865]
### Use-cases

Similar to #1651, I need to reference an existing workspace to attach it to a metastore.
Currently we can only hardcode the workspace ID.
### Proposal

```
data "databricks_workspace" "this" {
  name = "MyWorkspace"
}

resource "databricks_metastore_assignment" "this" {
  metastore_id = "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
  workspace_id = data.databricks_workspace.this.id
}
```
### References

#1651

**Comment by natbaker** [comment]
I believe you can get this from [databricks_mws_workspaces](https://github.com/databricks/terraform-provider-databricks/blob/master/docs/data-sources/mws_workspaces.md)

`
provider "databricks" {
  alias = "mws"
  account_id = "abcd-0123"
}

data "databricks_mws_workspaces" "all" {
    provider = databricks.mws 
}

resource "databricks_metastore_assignment" "this" {
  metastore_id = "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
  workspace_id = data.databricks_mws_workspaces.all.ids["MyWorkspace"]
}`

**Comment by alexott** [comment]
What about adding it to the `current_user` data source? We already have workspace URL there...

**Comment by riturr** [comment]
> I believe you can get this from [databricks_mws_workspaces](https://github.com/databricks/terraform-provider-databricks/blob/master/docs/data-sources/mws_workspaces.md)
> 
> ```
> provider "databricks" {
>   alias = "mws"
>   account_id = "abcd-0123"
> }
> 
> data "databricks_mws_workspaces" "all" {
>     provider = databricks.mws 
> }
> 
> resource "databricks_metastore_assignment" "this" {
>   metastore_id = "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
>   workspace_id = data.databricks_mws_workspaces.all.ids["MyWorkspace"]
> }
> ```

It doesn't seem to work for me, Am I missing something?. I'm using Azure Databricks.

When trying to resolve the `data` block Terraform throws an error not being able to parse the response: 

`Error: cannot read data mws workspaces: invalid character '<' looking for beginning of value
│ <!doctype html><html lang="en"><... (695 more bytes)`

I tried calling separately that endpoint (https://accounts.azuredatabricks.net/api/2.0/accounts/<account-id>/workspaces) and it returns:
```
<!doctype html>

<html lang="en">


<head>

    <meta charset="utf-8" />

    <meta name="viewport" content="width=device-width,initial-scale=1" />

    <meta name="theme-color" content="#000000" />

    <meta name="description" content="Databricks Sign in" />

    <title>Databricks - Sign in</title>
</head>
<body><noscript>You need to enable JavaScript to run this app.</noscript>

    <div id="login"></div>

</body>

</html>
```

**Comment by mgyucht** [comment]
I don't believe the `databricks_mws_*` APIs work in Azure. You probably need to use `azurerm_databricks_workspace` and reference the `id` attribute. See https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/data-sources/databricks_workspace for more details.

**Comment by alexott** [comment]
I think that the current config data source was added later.

btw,  this works on Azure now... (on account level)

```
data "databricks_mws_workspaces" "all" {
}
```

**Comment by guderkar** [comment]
> I think that the current config data source was added later.
> 
> btw, this works on Azure now... (on account level)
> 
> ```
> data "databricks_mws_workspaces" "all" {
> }
> ```

Hello any plans to expose additional attributes? Workspace URL would be handy. With opentofu it would allow us to fetch workspace URLs through account provider, dynamically initialize multiple workspace providers and manage resources in multiple workspaces.

One use case might be centralized permission management for workspaces including entititlements etc. which can be updated only with workspace provider :-)

**Comment by devlucasc** [comment]
Do we have any updates on this? Having the current config object exposing the current workspace id and name it will help a lot to know where the provider is connected.

**Comment by alexott** [comment]
@devlucasc unfortunately, there is no such information available via workspace level APIs

**Comment by devlucasc** [comment]
@alexott I know that is possible to extract the current id using the response header of any endpoint request through the header "x-databricks-org-id". Is it possible to expose this information as the workspace id?

**Comment by devlucasc** [comment]
If I not wrong, for Azure workspaces the return header is "X-Databricks-Azure-Workspace-Resource-Id"

**Comment by kishaningithub** [comment]
The pattern of a workspace url is just "https://`deployment name`.cloud.databricks.com" . I see `deployment_name` available in the list API. Will raise a PR to add the same
